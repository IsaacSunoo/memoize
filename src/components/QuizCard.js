import React, { Component } from 'react';

class QuizCard extends Component {

  compareAnswer = (e) => {
    let { incorrectQuestions, proto, userAnswers } = this.props;
    let input = e.target.value;
    console.log('quiz card', proto);

    const isIncorrectQuestion = incorrectQuestions.find(question => {
      return question.id === proto.id;
    })

    if (proto.prototype !== input && !isIncorrectQuestion) {
      incorrectQuestions.push(proto);
      userAnswers.push({ userAnswer: input, id: proto.id });
    } else if (proto.prototype === input && isIncorrectQuestion) {
      let idx = incorrectQuestions.indexOf(proto);
      incorrectQuestions.splice(idx, 1);
    }
  }

  render() {
    return (
      <div className='quiz-card'>
        <div className='question-container'>
          <p><strong>{this.props.proto.id}.</strong> {this.props.proto.answer}</p>
        </div>
        <form className='answer-container'>
          {this.props.options.map((option, idx) => {

            return <label key={idx + 1}><input key={idx + 1} type='radio' onClick={this.compareAnswer} value={option} className='quiz-btn' name={this.props.proto.id} />{option}</label>
          })}
        </form>
      </div>
    )
  }
}

export default QuizCard;