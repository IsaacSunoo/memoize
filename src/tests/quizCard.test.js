import React from 'react';
import QuizCard from '../components/QuizCard';
import { shallow } from 'enzyme';

const determineSolutionMock = jest.fn();

const mockData = [{
    "id": 1,
    "prototype": "shift()",
    "answer": "Removes and returns the first item in an Array",
    "incorrectChoices": [
        "pop()",
        "push()",
        "unshift()"
    ],
    "method": "mutator",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift"
},
{
    "id": 2,
    "prototype": "copyWithin()",
    "answer": "Shallow copies part of an array to another location in the same array and returns it, without modifying its size",
    "incorrectChoices": [
        "splice()",
        "findIndex()",
        "concat()"
    ],
    "method": "mutator",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/copyWithin"
}
]



const optionsArrMock = [{
    "id": 1,
    "prototype": "shift()",
    "answer": "Removes and returns the first item in an Array",
    "incorrectChoices": [
        "pop()",
        "push()",
        "unshift()"
    ],
    "method": "mutator",
    "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift"
},
    {
        "id": 2,
        "prototype": "copyWithin()",
        "answer": "Shallow copies part of an array to another location in the same array and returns it, without modifying its size",
        "incorrectChoices": [
            "splice()",
            "findIndex()",
            "concat()"
        ],
        "method": "mutator",
        "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/copyWithin"
    }];

const incorrectQuestionsMock = [];
const userAnswersMock = [];

describe('QuizCard', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <QuizCard
                key={mockData.id}
                proto={mockData[0]}
                options={optionsArrMock}
                incorrectQuestions={incorrectQuestionsMock}
                determineSolution={determineSolutionMock}
                userAnswers={userAnswersMock}
            />
        )
    })

    it('should match the snapshot with all data passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should invoke the compare answer function', () => {
        wrapper.find('.quiz-btn').at(0).simulate('click', { target: { value: 'pop()' } });
        // console.log('wrapper', wrapper);
        // console.log('wrapper prop', wrapper.instance().props.options)
        // console.log('incorrect Questions', wrapper.instance().props.incorrectQuestions)

    })
});