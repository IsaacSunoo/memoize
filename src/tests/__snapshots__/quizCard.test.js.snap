// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizCard should match the snapshot with all data passed in correctly 1`] = `
<div
  className="quiz-card"
>
  <div
    className="question-container"
  >
    <p>
      <strong>
        1
        .
      </strong>
       
      Removes and returns the first item in an Array
    </p>
  </div>
  <form
    className="answer-container"
  >
    <label
      key="1"
    >
      <input
        className="quiz-btn"
        key="1"
        name={1}
        onClick={[Function]}
        type="radio"
        value={
          Object {
            "answer": "Removes and returns the first item in an Array",
            "id": 1,
            "incorrectChoices": Array [
              "pop()",
              "push()",
              "unshift()",
            ],
            "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/shift",
            "method": "mutator",
            "prototype": "shift()",
          }
        }
      />
      <Component />
    </label>
    <label
      key="2"
    >
      <input
        className="quiz-btn"
        key="2"
        name={1}
        onClick={[Function]}
        type="radio"
        value={
          Object {
            "answer": "Shallow copies part of an array to another location in the same array and returns it, without modifying its size",
            "id": 2,
            "incorrectChoices": Array [
              "splice()",
              "findIndex()",
              "concat()",
            ],
            "link": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/copyWithin",
            "method": "mutator",
            "prototype": "copyWithin()",
          }
        }
      />
      <Component />
    </label>
  </form>
</div>
`;
